Review

Backss4 opened the pull request:
Utworzenie dokumentacji klasy IntLinkedList oraz dodanie "TODO".
Stworzenie diagramu UML.
Poprawienie i dopisanie testów jednostkowych.

Backss4 committed:
Dodane pliki:
-.gitignore

Backss4 committed:
Zadanie 1.3.2:
-poprawa formatowania plików
-sprawdzenie działania kombinacji klawiszy alt + ← oraz alt + →

Backss4 committed:
Merge branch 'format'
Zadanie 1.3.3:
-scalenie gałęzi master i format.

Backss4 committed:
Zadanie 1.3.5:
-poprawa nazw klas.

Backss4 committed:
Zadanie 1.3.5:
-poprawa nazw metod.

Backss4 committed:
Zadanie 1.3.7:
-poprawa formatowania nazw pól.

Backss4 committed:
Merge branch 'fields-naming'
Zadanie 1.3.8:
-scalenie master z fields-naming.

Backss4 committed:
Merge branch 'class-method-naming'
Zadanie 1.3.10:
-scalenie master i class-method-naming.

Backss4 committed:
1.4. Praca w podejściu feature branch

Temat: accessory i hermetyzacja

DefaultCountingOutRhymer: zmiana widoczności pól numbers i total

FIFORhymer: zmiana widoczności pola temp

HanoiRhymer: zmiana widoczności pola totalRejected

IntLinkedList: zmiana widoczności pól last i i

Node: zmiana widoczności pól prev i next, wygenerowanie accessorów dla tych pól, zmiana widoczności pola value i wygenerowanie gettera dla tego pola

Backss4 committed:
Merge pull request #2 from Backss4/accessory_i_hermetyzacja

1.4. Praca w podejściu feature branch

Backss4 committed:
1.5. Testy, diagramy, dokumentacja i *gwiazdki

RhymersDemo:
	rozbicie metody main. Stworzenie nowej metody testRhymers.

Node: przeniesienie klasy do klasy IntLinkedList.

Backss4 committed:
Poprawienie czytelnosci testów

Backss4 committed:
Dodanie testów dla klasy:
	IntLinkedList:
		-push
		-pop
		-top
		-isEmpty
	HanoiRhymer:
		-countIn
	FIFORhymer:
		-countOut

Backss4 committed:
Poprawki

Backss4 committed:
Dodanie szkieletu dokumentacji klasy IntLinkedList.

Napisanie TODO.

Backss4 committed:
Poprawki

Backss4 committed:
Dodanie UML

Backss4 committed:
Merge pull request #3 from Backss4/testy_diagramy_dokumentacja_i_gwiazdki

Testy diagramy dokumentacja i gwiazdki

Backss4 committed:
Poprawki

MarekZuchowski committed:
constants and annotations added

Backss4 committed:
Merge pull request #5 from Backss4/constants_n_annotations

constants and annotations added

Backss4 committed:
1.5 Testy diagramy dokumentacja i gwiazdki
Poprawki

kowallus commented on src/main/java/edu/kis/vh/nursery/DefaultCountingOutRhymer.java:
```diff
@@ -0,0 +1,45 @@
+    private static final int SIZE = 12;
+    private static final int INITIAL_VALUE = -1;
+    private static final int FAILURE = -1;
+    private static final int MAX_SIZE = 12;
```
This constant is related to SIZE. Changing only one leads to an error (inconsistency).

kowallus commented on src/main/java/edu/kis/vh/nursery/DefaultCountingOutRhymer.java:
```diff
@@ -0,0 +1,45 @@
+
+    private static final int SIZE = 12;
+    private static final int INITIAL_VALUE = -1;
+    private static final int FAILURE = -1;
```
Suffix _VALUE is more appropriate for this constants

kowallus commented on src/main/java/edu/kis/vh/nursery/DefaultCountingOutRhymer.java:
```diff
@@ -0,0 +1,45 @@
+    private static final int FAILURE = -1;
+    private static final int MAX_SIZE = 12;
+
+    private final int[] numbers = new int[SIZE];
```
This should be MAX_SIZE (or e.g., capacity)

Backss4 replied:
Fixed.

kowallus reviewed (commented):
5.5 pts

Specific issues inlined in the code.

Backss4 committed:
2.1.1 and 2.1.2
Ekstrakcja implementacji stosu z DefaultCountingOutRhymer do
IntArrayStack

2.1.1 i 2.1.2
Ekstrakcja implementacji stosu z DefaultCountingOutRhymer do
IntArrayStack

Backss4 committed:
2.2. Refaktoryzacja do wzorca bridge

Backss4 commented:
Changes in master branch.

Backss4 committed:
2.3. Fabryki, diagramy, testy, *gwiazdki
